src/actions/actions.ts

```diff
@@ -0,0 +1,32 @@
+"use server";
+
+import { getServerSession } from "next-auth";
+import { authOptions } from "@/src/app/api/auth/[...nextauth]/route";
+
+const { ObjectId } = require("mongodb");
+
+export async function isValidObjectId(id: string) {
+  return ObjectId.isValid(id) && new ObjectId(id).toString() === id;
+}
+
+export async function isAuthor(author: string) {
+  if (!author) return;
+
+  const session = await getServerSession(authOptions);
+
+  if (!session) return false;
+
+  if (author !== session.user?.email) return false;
+
+  return true;
+}
+
+export async function isReciever(reciever: string) {
```

Comment:
tutaj jak wyżej, nie wiadomo od razu że chodzi o maila

dodatkowo pytanie czy nie lepiej takich rzeczy sprawdzać po id?

Reply:
tak w ogóle to co te funkcje sprawdzają, isAuthor jest praktycznie tym samym co isReceiver a nawet nie ma innych argumentów pokroju "czego autorem jesteśmy"